palace/linalg/divfree.cpp

```diff
@@ -49,6 +49,42 @@ DivFreeSolver<VecType>::DivFreeSolver(
     int print)
 {
   BlockTimer bt(Timer::DIV_FREE);
+
+  // If no boundaries on the mesh have been marked, add a single degree of freedom
+  // constraint so the system for the projection is not singular. This amounts to enforcing
+  // a scalar potential of 0 at a point in space if it is otherwise completely
+  // unconstrained.
+  const auto *ptr_h1_bdr_tdof_lists = &h1_bdr_tdof_lists;
+  {
+    MFEM_VERIFY(
+        !h1_bdr_tdof_lists.empty(),
+        "Unexpected empty list of boundary true dofs for finite element space hierarchy!");
+    HYPRE_BigInt coarse_bdr_tdofs = h1_bdr_tdof_lists[0].Size();
+    MPI_Comm comm = h1_fespaces.GetFESpaceAtLevel(0).GetComm();
+    Mpi::GlobalSum(1, &coarse_bdr_tdofs, comm);
+    if (coarse_bdr_tdofs == 0)
+    {
+      MPI_Comm split_comm = MPI_COMM_NULL;
+      int color = (h1_fespaces.GetFESpaceAtLevel(0).GetTrueVSize() > 0) ? 0 : MPI_UNDEFINED;
+      MPI_Comm_split(comm, color, Mpi::Rank(comm), &split_comm);
+      if (split_comm != MPI_COMM_NULL && Mpi::Root(split_comm))
```

Comment:
So there's only one dof on each processor with more than one dof correct?

Reply:
Yeah this was a silly overcompilcation, thanks for pointing this out. Simplified in [ee3dbd7](https://github.com/awslabs/palace/pull/269/commits/ee3dbd73680c1c3109ccfb90569650aca1c60591).